vsintegration/src/FSharp.Editor/LanguageService/WorkspaceExtensions.fs

```diff
@@ -325,17 +323,20 @@ type Project with
                     |> Task.WhenAll
             else
                 for doc in documents do
-                    do!
-                        doc.FindFSharpReferencesAsync(symbol, (fun range -> onFound doc range), userOpName)
-                        |> RoslynHelpers.StartAsyncAsTask ct
+                    do! doc.FindFSharpReferencesAsync(symbol, (fun range -> onFound doc range), userOpName)
         }
 
-    member this.GetFSharpCompilationOptionsAsync(ct: CancellationToken) =
-        backgroundTask {
-            if this.IsFSharp then
-                match ProjectCache.Projects.TryGetValue(this) with
-                | true, result -> return result
-                | _ ->
+    member this.GetFSharpCompilationOptionsAsync() =
+        if not this.IsFSharp then
+            raise (OperationCanceledException("Project is not a FSharp project."))
```

Comment:
Unrelated to this PR - I just wonder how useful are these checks and why it can even happen here that the project wouldn't be an F# one.

Reply:
It can happen if there's a bug in CPS,  likely.